Use a Rtree query for tagMesh*Round

Summerdave opened the pull request:
This PR targets issue #293 by using boost::geometry for lookup.

floli commented on src/mapping/PetRadialBasisFctMapping.hpp:
```diff
@@ -770,24 +770,25 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshFirstRound()
-      v.tag();
+  }
+  for (auto& vert : insideVerts){
+    vert.tag();
```
Idea:

Move that loop (`for vert in insideVerts: vert.tag()`) into `if (not _basisFunction.hasCompactSupport)` and use the `make_function_output_iterator` to directly invoke `vert.tag()`. Got the idea?

Summerdave replied:
Clever.

floli commented on src/mapping/PetRadialBasisFctMapping.hpp:
```diff
@@ -825,18 +826,20 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshSecondRound()
+        insideVerts.push_back(&mesh->vertices()[idx]);
+      }));
+  for (auto& vert : insideVerts){
+    vert.tag();
```
Same may be possible here.

fsimonis commented:
@Summerdave Looks good.

Can you implement the tagging for the other mappings too? (The title of the PR suggests you change all of them.)

_For the record: I don't like the repetition of the type definition for Mesh::BoundingBox, but I get why you did it. Maybe we could add something like a `mesh_fwd.hpp` in the future?_

Summerdave commented:
> Can you implement the tagging for the other mappings too? (The title of the PR suggests you change all of them.)

The re-partitioning process for RBF mapping is based on @uekerman`s [thesis](https://mediatum.ub.tum.de/doc/1320661/document.pdf) (page 89). Right now, the partitioning for the other mappings works by computing a mapping and filtering this by whether the point _was used_. So for determining which points to tag a full mapping must be run anyway. @uekerman has also commented on this in #293.

Summerdave commented:
@floli Found a severe issue in the code which is now fixed. I also added unit tests to ensure `tagMeshFirstRound` and `tagMeshSecondRound` work correctly.
